src/bp/bpfs_recovery.ts

```diff
@@ -0,0 +1,85 @@
+import 'bluebird-global'
+// tslint:disable-next-line:ordered-imports
+import './sdk/rewire'
+// tslint:disable-next-line:ordered-imports
+import './common/polyfills'
+
+import chalk from 'chalk'
+import { FatalError } from 'errors'
+import fs from 'fs'
+import { mkdirpSync } from 'fs-extra'
+import path from 'path'
+import 'reflect-metadata'
+import { container } from './core/app.inversify'
+import Database from './core/database'
+import { GhostService } from './core/services'
+import { TYPES } from './core/types'
+
+export default async (argv, action) => {
+  let ghost: GhostService | undefined
+  let database: Database | undefined
+
+  try {
+    ghost = container.get<GhostService>(TYPES.GhostService)
+    database = container.get<Database>(TYPES.Database)
+
+    const { DATABASE_URL } = process.env
+
+    const useDbDriver = process.env.BPFS_STORAGE === 'database'
+    ghost.initialize(useDbDriver)
+
+    const dbType = DATABASE_URL && DATABASE_URL.toLowerCase().startsWith('postgres') ? 'postgres' : 'sqlite'
+    await database.initialize(dbType, DATABASE_URL)
+  } catch (err) {
+    console.error(chalk.red(`Error during initialization`), err)
+    return process.exit()
+  }
+
+  const { file, dest } = argv
+
+  if (argv.list) {
+    console.log(`Directory listing of ${path.join('data', argv.list)}
+=========================================`)
+
+    const files = await ghost.root().directoryListing(argv.list)
+    files.map(file => console.log(chalk.green(` - ${file}`)))
```

Comment:
`.map` is usually used to extract or transform the data of a container (array) in a new container (array). please use `.forEach` or `for ( const f of files)`